planner_cspace/src/planner_3d.cpp

```diff
@@ -2051,6 +2067,188 @@ class Planner3dNode
+      arrivable_map_.setParams(cc_, num_cost_estim_task_);
+      arrivable_map_.init(model_, dmp);
+      cm_local_esc_.copy_partially(
+          Astar::Vec(0, 0, 0), cm_rough_, local_origin, local_origin + local_size);
+      arrivable_map_.create(local_center, local_center);
+
+      // Find temporary goal
+      float cost_min = std::numeric_limits<float>::max();
+      Astar::Vec te_out;
+      const int esc_range_sq = esc_range_ * esc_range_;
+      const int esc_range_min_sq = esc_range_min_ * esc_range_min_;
+      for (Astar::Vec d(0, -esc_range_, 0); d[1] <= esc_range_; d[1]++)
+      {
+        for (d[0] = -esc_range_; d[0] <= esc_range_; d[0]++)
+        {
+          const int sqlen = d.sqlen();
+          if ((d[0] == 0 && d[1] == 0) || sqlen > esc_range_sq || sqlen < esc_range_min_sq)
+          {
+            continue;
+          }
+
+          Astar::Vec te(start_grid[0] + d[0], start_grid[1] + d[1], 0);
+          if ((unsigned int)te[0] >= (unsigned int)map_info_.width ||
+              (unsigned int)te[1] >= (unsigned int)map_info_.height)
+          {
+            continue;
+          }
+          if (!cm_rough_.validate(te, range_) || cm_rough_[te] >= 50)
+          {
+            continue;
+          }
+          const auto cost = cost_estim_cache_static_[te];
+          if (cost >= cost_min)
+          {
+            continue;
+          }
+
+          // Check arrivability
+          if (arrivable_map_[te - local_origin] == std::numeric_limits<float>::max())
+          {
+            continue;
+          }
+
+          if (te[0] == g_orig[0] && te[1] == g_orig[1] && cm_[g_orig] < 100)
+          {
+            // Original goal is in the temporary escape range and reachable
+            ROS_INFO("Original goal is reachable. Back to the original goal.");
+            goal_ = goal_raw_ = goal_original_;
+            escaping_ = false;
+            createCostEstimCache();
+            return;
+          }
+
+          // Calculate distance map gradient
+          float grad[2] = {0, 0};
+          for (Astar::Vec d(0, -1, 0); d[1] <= 1; d[1]++)
+          {
+            for (d[0] = -1; d[0] <= 1; d[0]++)
+            {
+              if (d[0] == 0 && d[1] == 0)
+              {
+                continue;
+              }
+
+              const auto p = te + d;
+              const auto cost2 = cost_estim_cache_static_[p];
+              if (cost2 == std::numeric_limits<float>::max())
+              {
+                continue;
+              }
+              const float cost_diff = cost2 - cost;
+              grad[0] += -cost_diff * d[0];
+              grad[1] += -cost_diff * d[1];
+            }
+          }
+          if (grad[0] == 0 && grad[1] == 0)
+          {
+            continue;
+          }
+          const float yaw = std::atan2(grad[1], grad[0]);
+          te[2] = static_cast<int>(yaw / map_info_.angular_resolution);
+
+          te.cycleUnsigned(map_info_.angle);
+          if (cm_[te] >= 50)
```

Comment:
What does the value 50 mean?

Reply:
It's intended to be same as the hard coded acceptable cost of start/goal position relocation (`searchAvailablePos`).
I changed it to a new parameter `relocation_acceptable_cost`.